lib/config/index.js

```diff
@@ -34,25 +34,20 @@ const configFilePath = path.resolve(appRootPath, process.env.CMD_CONFIG_FILE ||
 'config.json')
 const fileConfig = fs.existsSync(configFilePath) ? require(configFilePath)[env] : undefined
 
-let config = require('./default')
-merge(config, require('./defaultSSL'))
-merge(config, require('./oldDefault'))
-merge(config, debugConfig)
-merge(config, packageConfig)
-merge(config, fileConfig)
-merge(config, require('./oldEnvironment'))
-merge(config, require('./hackmdEnvironment'))
-merge(config, require('./environment'))
-merge(config, require('./dockerSecret'))
-
-if (['debug', 'verbose', 'info', 'warn', 'error'].includes(config.loglevel)) {
+let config = require('./convict')
+config.load(debugConfig)
+config.load(packageConfig)
+config.load(fileConfig)
+config.load(require('./dockerSecret'))
```

Comment:
If we break down the convict configuration, I think this would be the place to put it together. (not married to that, though if anything else is as readable).

Reply:
Convict doesn't work that way. You need bring the schema together before submitting it into convict. I would prefer to have once big schema for the top-level and then other schemas as part of the types for settings of components like CSP, LDAP, …

This should simplify config mangement over time and is listed as a next step above.